agent/lib/kontena/workers/node_info_worker.rb

```diff
@@ -72,17 +72,25 @@ def publish_node_info
     ##
     # @return [String, NilClass]
     def public_ip
-      Net::HTTP.get('whatismyip.akamai.com', '/')
+      if ENV['KONTENA_PUBLIC_IP'].to_s != ''
+        ENV['KONTENA_PUBLIC_IP'].to_s.strip
+      else
+        Net::HTTP.get('whatismyip.akamai.com', '/')
+      end
     rescue => exc
       error "Cannot resolve public ip: #{exc.message}"
       nil
     end
 
     # @return [String]
     def private_ip
-      ip = interface_ip(private_interface)
-      unless ip
-        ip = interface_ip('eth0')
+      if ENV['KONTENA_PRIVATE_IP'].to_s != ''
+        ip = ENV['KONTENA_PRIVATE_IP'].to_s.strip
+      else
+        ip = interface_ip(private_interface)
+        unless ip
+          ip = interface_ip('eth0')
```

Comment:
I would change this to single line conditional 
`ip = interface_ip('eth0') unless ip`

or it can be even
`ip = interface_ip(private_interface) || interface_ip('eth0')`